### ISSUE TITLE
Disable update of desired capacity for elastigroup with autoscaling

### ISSUE THREAD

**Issue by spa-87** [SungardAS/spotinst-lambda#6]
Hi, came time for autoscaling and I found a potentially danger place when autoscaling is used.

Suppose I launch elastigroup with target capacity of 2 instances. 
Workload grows and cluster scales-out...
Eventually I want to increase instance type. I go to CloudFormation and change corresponding parameter. But as there's set target capacity to 2 (from initial run), elastigroup is scaled-in to desired size what can lead to performance degradation and bad user experience. 

I think about some spotinst-lambda parameter which would let to block modification of target capacity. That parameter can be disabled by default what equals to the behavior we can see today. But if it's enabled, spotinst-labmda keeps unchanged elastigroup capacity.

**Comment by kmcgrath** [comment]
How is the cluster scaling out?  Are you going to the Spotinst console and changing desired state there?

I worry about this because it breaks the concept of a declared state.  If you need to grow capacity then the declaration should be changed (the CF template) which will update Spotinst accordingly.

If it is changed out-of-band then the template should be updated accordingly, if that is what is desired.

How is your desired state changing in this scenario?

**Comment by spa-87** [comment]
Yep, I totally agree that it's an incorrect behavior to change capacity via SpotInst UI and expect from CloudFormation to handle this case correctly. When we use CF template, all manual changes should be done via this service.

But I have a different use-case, I don't change any parameters manually. I want to use autoscaling. So I define scaling policies and SpotInst changes capacity automatically based on defined rules and CloudWatch metrics.

**Comment by kmcgrath** [comment]
I think I need to run some tests myself to get the full picture, but I want to be clear on what is happening.
1. CF has a template with desired state and autoscaling defined
2. CF creates the elastigroup with autoscaling
3. Spotinst scales the elastigroup (and the desired/target state chagnes?)
4. Updating CF is bringing the target back down to the original value

Does that look correct?

**Comment by spa-87** [comment]
Yes, all points are correct.

**Comment by kmcgrath** [comment]
A check needs to be performed to check if min, max or desired state has changed.  If not, remove them from the REST call to spotinst.
